trt_yolo_ros/src/trt_yolo_ros/trt_yolo_ros.py

```diff
@@ -109,26 +109,30 @@ def _write_message(self, detection_results, boxes, scores, classes):
     @timeit_ros
     def process_frame(self):
         """ Main function to process the frame and run the infererence """
-        if self.msg is None:
-            return
-        # Initialize detection results
-        detection_results = BoundingBoxes()
-        detection_results.header = self.msg.header
-        detection_results.image_header = self.msg.header
-        self.msg = None
-        rospy.logdebug("[trt_yolo_ros] processing frame")
-        boxes, classes, scores, visualization = self.model(self.image)
-        # and deleting the processed message from memmory
-
-        # construct message
-        self._write_message(detection_results, boxes, scores, classes)
-
-        # send message
+        # Deque the next image msg
+        current_msg = self.msg_queue.get()
+        current_image = None
+        # Convert to image to OpenCV format
         try:
-            rospy.logdebug("[trt_yolo_ros] publishing")
-            self._pub.publish(detection_results)
-            if self.publish_image:
-                self._pub_viz.publish(self._bridge.cv2_to_imgmsg(visualization, "bgr8"))
+            current_image = self._bridge.imgmsg_to_cv2(current_msg, "bgr8")
+            rospy.logdebug("[trt_yolo_ros] image converted for processing")
         except CvBridgeError as e:
-            print(e)
+            rospy.logdebug("Failed to convert image %s" , str(e))
+        # Initialize detection results
+        if current_image is not None:
+            boxes, classes, scores, visualization = self.model(current_image)
+            detection_results = BoundingBoxes()
+            detection_results.header = current_msg.header
+            detection_results.image_header = current_msg.header
+            rospy.logdebug("[trt_yolo_ros] processing frame")
```

Comment:
Can you move `logdebug` before `self.model` call? In that way if something will fail during the model run the last message would be `processing frame` which will make a bit more sense for debugging.